ui/selectors/confirm-transaction.js

```diff
@@ -217,10 +218,18 @@ export const transactionFeeSelector = function (state, txData) {
   const currentCurrency = currentCurrencySelector(state);
   const conversionRate = conversionRateSelector(state);
   const nativeCurrency = getNativeCurrency(state);
+  const metaSwapAverageGasEstimate = getAveragePriceEstimateInHexWEI(state);
 
-  const {
-    txParams: { value = '0x0', gas: gasLimit = '0x0', gasPrice = '0x0' } = {},
+  const { txParams: { value = '0x0', gas: gasLimit = '0x0' } = {} } = txData;
+
+  // if the gas price from our infura endpoint is null or undefined
+  // use the metaswap average price estimation as a fallback
+  let {
+    txParams: { gasPrice },
   } = txData;
+  if (!gasPrice) {
+    gasPrice = metaSwapAverageGasEstimate || '0x0';
+  }
```

Comment:
These changes looks good. 

Q: Please check the function `updateTxDataAndCalculate` in`confirm-transaction.duck` as it seems to be getting the `gasPrice` from infura as well. (This function is called from confirm-transaction.component>>componentDidUpdate>>setTransactionToConfirm).

Reply:
Good catch! Thanks Niranjana. I'll add the fallback here!